File: ExtremeStudio/ExtremeParser/Types/DefinesStruct.vb
Imports System.Text.RegularExpressions

Public Structure DefinesStruct

    'No properties for speed.
    Public DefineName As String
    Public DefineValue As String
    Public RegexMatch As Match

    Public Sub New(_defineName As String, _defineValue As String, _RegexMatch As Match)
        DefineName = _defineName
        DefineValue = _defineValue
        RegexMatch = _RegexMatch
    End Sub

    Public Shared Operator =(ByVal first As DefinesStruct, second As DefinesStruct)
        If first.DefineName = second.DefineName And first.DefineValue = second.DefineValue Then
            Return True
        End If
        Return False
    End Operator
    Public Shared Operator <>(ByVal first As DefinesStruct, second As DefinesStruct)
        If first = second Then
            Return False
        Else
            Return True
        End If
    End Operator
End Structure

File: ExtremeStudio.Parser/Types/DefinesStruct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtremeStudio.Parser.Types
{
    public struct DefinesStruct
    {

        //No properties for speed.
        public string DefineName;
        public string DefineValue;

        public DefinesStruct(string defName, string defValue)
        {
            DefineName = defName;
            DefineValue = defValue;
        }

        public static object operator ==(DefinesStruct first, DefinesStruct second)
        {
            if (first.DefineName == second.DefineName && first.DefineValue == second.DefineValue)
            {
                return true;
            }
            return false;
        }
        public static object operator !=(DefinesStruct first, DefinesStruct second)
        {
            if ((bool)(first == second))
            {
                return false;
            }
            else
            {
                return true;
            }
        }

        public override bool Equals(object o)
        {
            throw new NotImplementedException();
        }
        public override int GetHashCode()
        {
            throw new NotImplementedException();
        }
    }
}
